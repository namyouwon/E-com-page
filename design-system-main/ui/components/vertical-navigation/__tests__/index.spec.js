// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import {
  NavVerticalFieldset,
  NavVerticalItemRadio
} from '../radio-group/example';
import {
  NavVerticalItem,
  NavVerticalSection,
  NavVertical
} from '../list/example';

const { matchesMarkup } = createHelpers(__dirname);

describe('Vertical Nav Radio Group', () => {
  it('renders a radio button', () =>
    matchesMarkup(<NavVerticalItemRadio id="test1" />));

  it('renders a checked radio button', () =>
    matchesMarkup(<NavVerticalItemRadio id="test1" checked />));

  it('renders a fieldset', () =>
    matchesMarkup(
      <NavVerticalFieldset>
        <NavVerticalItemRadio id="test1">test1</NavVerticalItemRadio>
      </NavVerticalFieldset>
    ));

  it('renders a shaded fieldset', () =>
    matchesMarkup(
      <NavVerticalFieldset shaded>
        <NavVerticalItemRadio id="test1">test1</NavVerticalItemRadio>
      </NavVerticalFieldset>
    ));

  it('renders a shaded fieldset with a checked item', () =>
    matchesMarkup(
      <NavVerticalFieldset shaded>
        <NavVerticalItemRadio checked id="test1">
          test1
        </NavVerticalItemRadio>
      </NavVerticalFieldset>
    ));

  it('renders a compact fieldset', () =>
    matchesMarkup(
      <NavVerticalFieldset compact>
        <NavVerticalItemRadio id="test1">test1</NavVerticalItemRadio>
      </NavVerticalFieldset>
    ));
});

describe('Vertical Nav List Group', () => {
  it('renders a nav wrapper', () => matchesMarkup(<NavVertical />));

  it('renders a shaded nav wrapper', () =>
    matchesMarkup(<NavVertical isShaded />));

  it('renders a compact nav wrapper', () =>
    matchesMarkup(<NavVertical isCompact />));

  it('renders a list item', () =>
    matchesMarkup(<NavVerticalItem>Item One</NavVerticalItem>));

  it('renders an active list item', () =>
    matchesMarkup(<NavVerticalItem active> Item One</NavVerticalItem>));

  it('renders a list section', () =>
    matchesMarkup(<NavVerticalSection headerId="header1" title="Folders" />));

  it('renders a collapsed list section', () =>
    matchesMarkup(
      <NavVerticalSection
        headerId="header1"
        title="Folders"
        listId="list1"
        collapsed
      />
    ));

  it('renders an expanded list section', () =>
    matchesMarkup(
      <NavVerticalSection
        headerId="header1"
        title="Folders"
        listId="list1"
        expanded
      />
    ));

  it('renders a div instead of a list for radio sets', () =>
    matchesMarkup(<NavVerticalSection isNavVerticalRadioSet />));
});
