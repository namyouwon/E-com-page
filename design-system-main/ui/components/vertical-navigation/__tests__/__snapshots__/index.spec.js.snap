// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertical Nav List Group renders a collapsed list section 1`] = `
<div
  className="slds-nav-vertical__overflow"
>
  <button
    aria-controls="list1"
    aria-expanded="false"
    className="slds-button slds-button_reset slds-nav-vertical__action slds-nav-vertical__action_overflow"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
      />
    </svg>
    <span
      className="slds-nav-vertical__action-text"
    >
      Show More
      <span
        className="slds-assistive-text"
      >
        Folders
      </span>
    </span>
  </button>
  <div
    className="slds-hide"
    id="list1"
  >
    <ul
      aria-describedby="header1"
    />
  </div>
</div>
`;

exports[`Vertical Nav List Group renders a compact nav wrapper 1`] = `
<nav
  aria-label="Sub page"
  className="slds-nav-vertical slds-nav-vertical_compact"
/>
`;

exports[`Vertical Nav List Group renders a div instead of a list for radio sets 1`] = `
<div
  className="slds-nav-vertical__section"
>
  <div />
</div>
`;

exports[`Vertical Nav List Group renders a list item 1`] = `
<li
  className="slds-nav-vertical__item"
>
  <a
    className="slds-nav-vertical__action"
    href="#"
    onClick={[Function]}
  >
    Item One
  </a>
</li>
`;

exports[`Vertical Nav List Group renders a list section 1`] = `
<div
  className="slds-nav-vertical__section"
>
  <h2
    className="slds-nav-vertical__title"
    id="header1"
  >
    Folders
  </h2>
  <ul
    aria-describedby="header1"
  />
</div>
`;

exports[`Vertical Nav List Group renders a nav wrapper 1`] = `
<nav
  aria-label="Sub page"
  className="slds-nav-vertical"
/>
`;

exports[`Vertical Nav List Group renders a shaded nav wrapper 1`] = `
<nav
  aria-label="Sub page"
  className="slds-nav-vertical slds-nav-vertical_shade"
/>
`;

exports[`Vertical Nav List Group renders an active list item 1`] = `
<li
  className="slds-nav-vertical__item slds-is-active"
>
  <a
    aria-current={true}
    className="slds-nav-vertical__action"
    href="#"
    onClick={[Function]}
  >
     Item One
  </a>
</li>
`;

exports[`Vertical Nav List Group renders an expanded list section 1`] = `
<div
  className="slds-nav-vertical__overflow"
>
  <button
    aria-controls="list1"
    aria-expanded="true"
    className="slds-button slds-button_reset slds-nav-vertical__action slds-nav-vertical__action_overflow"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#chevronright"
      />
    </svg>
    <span
      className="slds-nav-vertical__action-text"
    >
      Show Less
      <span
        className="slds-assistive-text"
      >
        Folders
      </span>
    </span>
  </button>
  <div
    className="slds-show"
    id="list1"
  >
    <ul
      aria-describedby="header1"
    />
  </div>
</div>
`;

exports[`Vertical Nav Radio Group renders a checked radio button 1`] = `
<span
  className="slds-nav-vertical__item"
>
  <input
    defaultChecked={true}
    id="test1"
    name="unique-id-example"
    type="radio"
    value="test1"
  />
  <label
    className="slds-nav-vertical__action"
    htmlFor="test1"
  >
    <span
      className="slds-nav-vertical_radio-faux"
    />
  </label>
</span>
`;

exports[`Vertical Nav Radio Group renders a compact fieldset 1`] = `
<fieldset
  className="slds-nav-vertical slds-nav-vertical_compact"
>
  <legend
    className="slds-nav-vertical__title"
  >
    Reports
  </legend>
  <span
    className="slds-nav-vertical__item"
  >
    <input
      id="test1"
      name="unique-id-example"
      type="radio"
      value="test1"
    />
    <label
      className="slds-nav-vertical__action"
      htmlFor="test1"
    >
      <span
        className="slds-nav-vertical_radio-faux"
      >
        test1
      </span>
    </label>
  </span>
</fieldset>
`;

exports[`Vertical Nav Radio Group renders a fieldset 1`] = `
<fieldset
  className="slds-nav-vertical"
>
  <legend
    className="slds-nav-vertical__title"
  >
    Reports
  </legend>
  <span
    className="slds-nav-vertical__item"
  >
    <input
      id="test1"
      name="unique-id-example"
      type="radio"
      value="test1"
    />
    <label
      className="slds-nav-vertical__action"
      htmlFor="test1"
    >
      <span
        className="slds-nav-vertical_radio-faux"
      >
        test1
      </span>
    </label>
  </span>
</fieldset>
`;

exports[`Vertical Nav Radio Group renders a radio button 1`] = `
<span
  className="slds-nav-vertical__item"
>
  <input
    id="test1"
    name="unique-id-example"
    type="radio"
    value="test1"
  />
  <label
    className="slds-nav-vertical__action"
    htmlFor="test1"
  >
    <span
      className="slds-nav-vertical_radio-faux"
    />
  </label>
</span>
`;

exports[`Vertical Nav Radio Group renders a shaded fieldset 1`] = `
<fieldset
  className="slds-nav-vertical slds-nav-vertical_shade"
>
  <legend
    className="slds-nav-vertical__title"
  >
    Reports
  </legend>
  <span
    className="slds-nav-vertical__item"
  >
    <input
      id="test1"
      name="unique-id-example"
      type="radio"
      value="test1"
    />
    <label
      className="slds-nav-vertical__action"
      htmlFor="test1"
    >
      <span
        className="slds-nav-vertical_radio-faux"
      >
        test1
      </span>
    </label>
  </span>
</fieldset>
`;

exports[`Vertical Nav Radio Group renders a shaded fieldset with a checked item 1`] = `
<fieldset
  className="slds-nav-vertical slds-nav-vertical_shade"
>
  <legend
    className="slds-nav-vertical__title"
  >
    Reports
  </legend>
  <span
    className="slds-nav-vertical__item"
  >
    <input
      defaultChecked={true}
      id="test1"
      name="unique-id-example"
      type="radio"
      value="test1"
    />
    <label
      className="slds-nav-vertical__action"
      htmlFor="test1"
    >
      <span
        className="slds-nav-vertical_radio-faux"
      >
        test1
      </span>
    </label>
  </span>
</fieldset>
`;
